Added conforming datatypes & copatibility info (#3293)
Codecademy/docs: content/swift/concepts/protocols/terms/encodable/encodable.md

---

> Title: 'Encodable'

Reviewer: The changes to the metadata break the entry and should be removed. You should review the rules on metadata [here](https://github.com/Codecademy/docs/blob/main/documentation/style-guide.md#metadata) before attempting to edit another entry's metadata. Please revert these changes to the metadata.

Author: I restored the whole section. Sorry for distracting the intended structure. I was focused on the layout and unfortunately did not realize breaking the entry's metadata.

Description: 
 - 'A type that can convert itself into an external representation.'
 - 'handles data encoding tasks'
Subjects:
  - 'iOS 8.0+ | iPadOS 8.0+ | macOS 10.10+ | Mac Catalyst 13.0+ | tvOS 9.0+ | watchOS 2.0+ | visionOS 1.0 + Beta'
Tags:
  - 'Protocols'
  - 'APIs'
  - 'encoding'
CatalogContent:
  - 'learn-swift'
  - 'paths/build-ios-apps-with-swiftui'
---

> **`Encodable`** is a protocol that enables a type's data to be encoded and decoded. 

Reviewer: ```suggestion
**`Encodable`** is a protocol that enables a type's data to be encoded for use with Application Programming Interfaces ([APIs](https://www.codecademy.com/resources/docs/general/api)).
```
I would like to revert this change.

> > **Note:** A type that can encode itself to an external representation. The protocol is related to the following protocols
> - Codeable
> - Decodeable
> - CodingKey
> - CodingKeyRepresentable
> - CodingUserInfoKey

Reviewer: ```suggestion
> **Note:** This protocol is related to the following protocols:
> - Codeable
> - Decodeable
> - CodingKey
> - CodingKeyRepresentable
> - CodingUserInfoKey
```

- Codeable
- Decodeable
- CodingKey
- CodingKeyRepresentable
- CodingUserInfoKey

> > **Note:** If data also needs to be decoded, it's convenient to use the `Codable` protocol because it includes both the `Encodable` and `Decodable` protocols.

Reviewer: I'd like you not to delete this line.

Author: I added the line again in the document.

Reviewer 2: @BBrandl948 It's _best left as a note_ to differentiate its content from the rest of the introduction, since it's something the reader should be aware of outside the term

---

> Title: 'Encodeable'

Reviewer: ```suggestion
Title: 'Encodable'
```

Description: 'Enables object data to be encoded for use with Application Programming Interfaces.'
 
Subjects:
  - 'iOS'

>   - 'MobileDevelopment'

Reviewer: ```suggestion
  - 'Mobile Development'
```

  - 'paths/build-ios-apps-with-swiftui'
---

**`Encodable`** is a protocol that enables a type's data to be encoded and for use with Application Programming Interfaces ([APIs](https://www.codecademy.com/resources/docs/general/api)). If data also needs to be decoded, it's convenient to use the `Codable` protocol because it includes both the `Encodable` and `Decodable` protocols.

The protocol is related to the following protocols:

- Codeable: 
- [Decodeable](https://www.codecademy.com/resources/docs/swift/protocols/decodable):
- CodingKey:

> - CodingKeyRepresentable:

Reviewer 2: @BBrandl948 Can you include a brief sentence for each about how it is related

Author: I will try to

Author: I hope my suggestions meet your intentions. Committed all changes

  - 'paths/build-ios-apps-with-swiftui'
---

> **`Encodable`** is a protocol that enables a type's data to be encoded and for use with Application Programming Interfaces ([APIs](https://www.codecademy.com/resources/docs/general/api)).
> **Note:** If data also needs to be decoded, it's convenient to use the `Codable` protocol because it includes both the `Encodable` and `Decodable` protocols.
> > **Note:** A type that can encode itself to an external representation. The protocol is related to the following protocols

Reviewer: ```suggestion
**`Encodable`** is a protocol that enables a type's data to be encoded and for use with Application Programming Interfaces ([APIs](https://www.codecademy.com/resources/docs/general/api)). If data also needs to be decoded, it's convenient to use the `Codable` protocol because it includes both the `Encodable` and `Decodable` protocols.

The protocol is related to the following protocols:
```

}
```

> ## Conforming Types:

Reviewer: This entire section is a large infodump in the middle of the entry. Docs entries are supposed to be relatively short and easy to digest. I don't think this list is adding anything to the entry. I'd remove it.

Author: I deleted all lines. I think, that I had too little knowledge about Codecademy docs and its intentions when I started to work on this article. So I simply added Information, that seemed to be important to me. It was a demanded exercise in the iOS development career path and I could not find out in the description what exactly should be covered in the articles. So I simply jumped in.

Author: I made all changes in my local repo. In case that nearly all of my adds do not really match, I would suggest to simply delete my branch. If you do not agree with me please let me know what to do next. Thank you for your understanding.
